tools/comdb2ar/serialise.cpp

```diff
@@ -893,36 +950,122 @@ void serialise_database(
     std::ostringstream manifest;
-    manifest << "# Manifest for serialisation of " << dbname << std::endl;
-    if(support_files_only) {
-        manifest << "SupportFilesOnly" << std::endl;
-        if (origlrlname != "")
-        {
-           manifest <<"OrigLrlFile "<< origlrlname<< std::endl;
+
+    // Non-incremental mode or increment creation mode
+    if(!incr_gen){
+        manifest << "# Manifest for serialisation of " << dbname << std::endl;
+        if(support_files_only) {
+            manifest << "SupportFilesOnly" << std::endl;
+            if (origlrlname != "")
+            {
+               manifest <<"OrigLrlFile "<< origlrlname<< std::endl;
+            }
         }
-    }
 
-    for(std::list<FileInfo>::const_iterator
-            it = data_files.begin();
-            it != data_files.end();
-            ++it) {
-            write_manifest_entry(manifest, *it);
-    }
+        for(std::list<FileInfo>::const_iterator
+                it = data_files.begin();
+                it != data_files.end();
+                ++it) {
+                write_manifest_entry(manifest, *it);
+        }
+
+        // Find a recovery point after the copy, and record it in the manifest
+        if (!support_files_only) {
+            std::clog << "logdelete version " << log_holder->version() << std::endl;
+            if (log_holder->version() >= 3) {
+                std::string recovery_options = log_holder->recovery_options();
+                if (!recovery_options.empty()) {
+                    manifest << "Option " << recovery_options <<std::endl;
+                }
+            }
+        }
+
+        // Serialise the manifest file
+        serialise_string("MANIFEST", manifest.str());
+
+    // Incremental Mode
+    } else {
+        manifest << "# Manifest for serialisation of increment produced on "
+            << getDTString() << std::endl;
+
+
+        for(std::list<FileInfo>::const_iterator
+                it = data_files.begin();
+                it != data_files.end();
+                ++it) {
+
+            std::vector<uint32_t> pages_list;
+            ssize_t data_size = 0;
+
+            // Diff the page checksums for each file to find what has been changed
+            if(compare_checksum(*it, incr_path, pages_list, &data_size, incr_files)) {
+                // If pages list is empty but compare_checksum returned true, it's a new file
+                if(pages_list.empty()){
+                    new_files.push_back(*it);
+                    write_incr_manifest_entry(manifest, *it, pages_list);
+                // Keep track of the file name and the list of pages to be backed-up
+                } else {
+                    incr_data_files.push_back(*it);
+                    page_number_vec.push_back(pages_list);
+                    write_incr_manifest_entry(manifest, *it, pages_list);
+                    total_data_size += data_size;
+                }
```

Comment:
This works.  Because you are recording the pages in a separate step, other pages could be modified- this is okay, though: logfiles are being held so these changes will be restored in recovery.